### PR TITLE
Fix added vMI configuration to support ARM

### PR DESCRIPTION
**Pull request by hellocloudnative**
## Release note:

```release-note
NONE
```

### CODE REVIEW

**Comment by hellocloudnative** [comment]
@rmohr   Hello, what do you think?

**Comment by rmohr** [comment]
Hm, I would prefer if we can keep the memory footprint small in general. But not everyone has that much of memory for their dev clusters. I agree it is a little bit a tricky situation. Let's just take in the extra yaml for now.

**Comment by rmohr** [comment]
/test all

@hellocloudnative I think you will have to format some code and the tests will fail on that.

**Comment by hellocloudnative** [comment]
Hello, what do I need to do?  @rmohr

**Comment by zhlhahaha** [comment]
> Hello, what do I need to do? @rmohr

Here is the error log. You can see the style errors.
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6639/pull-kubevirt-generate/1455453491936890880

Or you can try to run `make generate-verify` in your local machine.

File: tools/vms-generator/utils/utils.go [hunk 1]
```diff
@@ -1051,3 +1052,14 @@ func GetVMIMacvtap() *v1.VirtualMachineInstance {
 	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{{Name: macvtapNetworkName, InterfaceBindingMethod: v1.InterfaceBindingMethod{Macvtap: macvtap}}}
 	return vmi
 }
+
+func GetVMIARM() *v1.VirtualMachineInstance {
+	vmi := getBaseVMI(VmiARM)
```
Comment (zhlhahaha) [inline tools/vms-generator/utils/utils.go:1057]: Would you mind adding a comment here to note why Arm64 VMs need 256Mi memory, something like
```
// The minimum memory for UEFI boot on Arm64 is 256Mi
```

**Comment by hellocloudnative** [comment]
Ok, I'll try @zhlhahaha

File: tools/vms-generator/utils/utils.go (hunk 1 above)
Comment (hellocloudnative) [reply tools/vms-generator/utils/utils.go:1057]: Ok, sure

**Comment by zhlhahaha** [comment]
> /retest

Have you run `make generate-verify` in your machine?

**Comment by hellocloudnative** [comment]
@zhlhahaha  Hello, now you can trigger the test

**Comment by hellocloudnative** [comment]
I ran make generate

**Comment by hellocloudnative** [comment]
image
Do I have to submit these as well?

**Comment by zhlhahaha** [comment]
> @zhlhahaha Hello, now you can trigger the test

I can see there are still some formatting issues, like using spaces rather than taps for indentation in the go file.

```
no changes added to commit (use "git add" and/or "git commit -a")
diff --git a/examples/vmi-arm.yaml b/examples/vmi-arm.yaml
index 7b970b755..74d3e2421 100644
--- a/examples/vmi-arm.yaml
+++ b/examples/vmi-arm.yaml
@@ -24,7 +24,7 @@ spec:
   terminationGracePeriodSeconds: 0
   volumes:
   - containerDisk:
-      image: registry:5000/kubevirt/cirros-container-disk-demo:devel 
+      image: registry:5000/kubevirt/cirros-container-disk-demo:devel
     name: containerdisk
   - cloudInitNoCloud:
       userData: |
@@ -35,4 +35,3 @@ spec:
   - emptyDisk:
       capacity: 2Gi
     name: emptydisk
-
diff --git a/tools/vms-generator/utils/utils.go b/tools/vms-generator/utils/utils.go
index 204e2540f..ead8e89da 100644
--- a/tools/vms-generator/utils/utils.go
+++ b/tools/vms-generator/utils/utils.go
@@ -57,7 +57,7 @@ const (
 	VmiMultusMultipleNet = "vmi-multus-multiple-net"
 	VmiHostDisk          = "vmi-host-disk"
 	VmiGPU               = "vmi-gpu"
-        VmiARM               = "vmi-arm"
+	VmiARM               = "vmi-arm"
 	VmiMacvtap           = "vmi-macvtap"
 	VmTemplateFedora     = "vm-template-fedora"
 	VmTemplateRHEL7      = "vm-template-rhel7"
@@ -1062,4 +1062,3 @@ func GetVMIARM() *v1.VirtualMachineInstance {
 	return vmi
 
 }
-
diff --git a/tools/vms-generator/vms-generator.go b/tools/vms-generator/vms-generator.go
index ca56a0760..ae65714d6 100644
--- a/tools/vms-generator/vms-generator.go
+++ b/tools/vms-generator/vms-generator.go
@@ -94,7 +94,7 @@ pciHostDevices:
 		utils.VmiGPU:               utils.GetVMIGPU(),
 		utils.VmiMacvtap:           utils.GetVMIMacvtap(),
 		utils.VmiKernelBoot:        utils.GetVMIKernelBoot(),
-                utils.VmiARM:               utils.GetVMIARM(),
+		utils.VmiARM:               utils.GetVMIARM(),
 	}
 
 	var vmireplicasets = map[string]*v1.VirtualMachineInstanceReplicaSet{
make: *** [Makefile:50: generate-verify] Error 1
+ EXIT_VALUE=2
+ set +o xtrace
Cleaning up after docker in docker.
```

**Comment by hellocloudnative** [comment]
image
Do I have to submit these as well? @zhlhahaha

**Comment by hellocloudnative** [comment]
I feel strange, my machine is normal
image
@zhlhahaha

**Comment by zhlhahaha** [comment]
> /retest @zhlhahaha That should be fine now, thank you very much

There are other two comments.
Especially, `The indentation is not aligned` will lead to testing failure.

**Comment by hellocloudnative** [comment]
What are the other two?

**Review by zhlhahaha — CHANGES_REQUESTED** [review]

**Comment by zhlhahaha** [comment]
> What are the other two? @zhlhahaha

Sorry, I forgot to submit the review, now can you see my comments?

File: examples/vmi-arm.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - disk:
+          bus: virtio
+        name: emptydisk
+    resources:
+      requests:
+        memory: 256Mi
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - containerDisk:
+      image: registry:5000/kubevirt/cirros-container-disk-demo:devel
```
[the remark this answers is not in the document]
Comment (hellocloudnative) [reply examples/vmi-arm.yaml:27]: This YARML file is automatically generated, should there be no problem?

**Comment by hellocloudnative** [comment]
I can see it now

**Comment by zhlhahaha** [comment]
Ok, let's give it a try
/ok-to-test

File: tools/vms-generator/utils/utils.go [hunk 3]
```diff
@@ -1051,3 +1052,14 @@ func GetVMIMacvtap() *v1.VirtualMachineInstance {
...
+// The minimum memory for UEFI boot on Arm64 is 256Mi
+func GetVMIARM() *v1.VirtualMachineInstance {
+	vmi := getBaseVMI(VmiARM)
+	vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("256Mi")
+	addContainerDisk(&vmi.Spec, fmt.Sprintf("%s/%s:%s", DockerPrefix, imageCirros, DockerTag), busVirtio)
+	addNoCloudDisk(&vmi.Spec)
+	addEmptyDisk(&vmi.Spec, "2Gi")
+	return vmi
+}
+
```
Comment (zhlhahaha) [inline tools/vms-generator/utils/utils.go:1065]: Remove the blank line

**Review by zhlhahaha — CHANGES_REQUESTED** [review]

File: tools/vms-generator/utils/utils.go (hunk 3 above)
Comment (hellocloudnative) [reply tools/vms-generator/utils/utils.go:1065]: ok

Comment (hellocloudnative) [reply tools/vms-generator/utils/utils.go]: Let's try it again

**Comment by hellocloudnative** [comment]
@zhlhahaha  hello.Do you have a Slack account and can I add you as a friend?

**Comment by zhlhahaha** [comment]
> @zhlhahaha hello.Do you have a Slack account and can I add you as a friend?

You can find me in the Slack channel [#virtualization @ kubernetes.slack.com](https://kubernetes.slack.com/?redir=%2Farchives%2FC8ED7RKFE)

My slack name is Howard Zhang

**Comment by hellocloudnative** [comment]
ok.thanks.

**Comment by rmohr** [comment]
/lgtm
/approve

Thanks :+1:

**Pull request merged**